search.go

```diff
@@ -481,5 +481,8 @@ func (sr *SearchResult) Merge(other *SearchResult) {
 	if other.MaxScore > sr.MaxScore {
 		sr.MaxScore = other.MaxScore
 	}
+	if len(sr.Facets) == 0 && len(other.Facets) != 0 {
```

Comment:
it could be len 0, but not nil, and we would create yet another map

would it be simpler if sr.Facets == nil && len(other.Facets) != 0 to simply use the right-hand-side of the expression, sr.Facets = other.Facets?  or does this introduce new problems?

Reply:
done!